### PR TITLE
Feat: :sparkles: Config API for Combat Tracker

### PR DESCRIPTION
**Pull request by aMediocreDad**
## Summary

- feat: ✨ Add Schema and default config
- feat: ✨ Resolve default config and implementation
- chore: 🔧 Example (Remove before merge)

## Checklist
- [x] Base implementation
- [x] Expose CONFIG via hook (see #9)
- [x] Implement DOM updates
- [x] Handle all basic use cases

### PR Type
- [x] This PR adds a new feature that is not an open feature request.

### CODE REVIEW

File: src/module/handlebars.js [hunk 1]
```diff
@@ -84,6 +84,10 @@ function registerHandlebarsHelpers() {
     return (a / b) * 100;
   });
 
+  Handlebars.registerHelper('getProperty', function (obj, key) {
```
Comment (Stefouch) [inline src/module/handlebars.js:87]: This is redundant with Handlebars **built-in** helper  `lookup`
https://handlebarsjs.com/guide/builtin-helpers.html#lookup

e.g:
```hbs
{{lookup obj key}}
```

File: src/templates/sidebar/combat-tracker.hbs [hunk 2]
```diff
@@ -1,124 +1,147 @@
+{{! Defines a partial that is used below to iterate over various button contexts}}
+{{#*inline "buttonPartial"}}
+	{{#each buttons}}
+		<a class="combatant-control {{#if (lookup ../combatant property)}}active{{/if}}" id={{id}} data-event={{eventName}} data-property="{{property}}" data-tooltip="{{localize tooltip}}">
+			<i class="fas {{icon}}"></i></a>
+	{{/each}}
+{{/inline}}
```
Comment (Stefouch) [inline src/templates/sidebar/combat-tracker.hbs:7]: Small thought: Wouldn't this be better placed into the `./src/module/handlebars.js` file ?

File: src/templates/sidebar/combat-tracker.hbs [hunk 3]
```diff
@@ -1,124 +1,147 @@
...
-                <i class="fas fa-caret-right"></i>
-            </a>
-            {{/if}}
-            <a class="combat-control" title="{{localize 'COMBAT.Delete'}}" data-control="endCombat" {{#unless combatCount}}disabled{{/unless}}>
-                <i class="fas fa-trash"></i>
-            </a>
-        </nav>
-        {{/if}}
+	<header id="combat-round">
+		{{#if user.isGM}}
```
Comment (Stefouch) [inline src/templates/sidebar/combat-tracker.hbs:11]: Is `.isGM` the only property we read from `user` ?

File: src/yze-combat.js [hunk 4]
```diff
@@ -64,8 +64,30 @@ Hooks.once('ready', async () => {
...
 
   console.log('YZEC | READY!');
+
+  // TODO Remove this example before merge
+
+  // This listens for a message from the client called when the user clicks the slow button in the combat tracker.
+  Hooks.on(`${MODULE_NAME}.slow-action-button-clicked`, data => {
+    console.log('YZEC | Event', data);
+    // The hook supplies a socket emit function that can be used to send a message to the server/other clients.
+    data.emit({ forGmOnly: 'secret lover' });
```
Comment (Stefouch) [inline src/yze-combat.js:74]: Love you too!

File: src/yze-combat.js [hunk 5]
```diff
@@ -64,8 +64,30 @@ Hooks.once('ready', async () => {
...
+  });
+  // This listens for a message from the client called when the user clicks the Duplicate button in the combat tracker.
+  Hooks.on(`${MODULE_NAME}.combat-tracker-duplicate-clicked`, data => {
```
Comment (Stefouch) [inline src/yze-combat.js:77]:
```suggestion
  Hooks.on(`${MODULE_NAME}.combat-tracker-duplicate-button-clicked`, data => {
```
Suggestion for staying consistent with the above one.

File: src/templates/sidebar/combat-tracker.hbs (hunk 2 above)
Comment (Stefouch) [reply src/templates/sidebar/combat-tracker.hbs:7]: Resolve: better design as used nowhere else

File: src/yze-combat.js (hunk 4 above)
Comment (Stefouch) [reply src/yze-combat.js:74]: Atropos design

File: src/templates/sidebar/combat-tracker.hbs (hunk 3 above)
Comment (Stefouch) [reply src/templates/sidebar/combat-tracker.hbs:11]: Atropos design

**Review by DrOgres — APPROVED** [review]

**Pull request merged**